tests/integration/pyramid_tests.py

```diff
@@ -51,6 +60,22 @@ def setUp(self):
         configurator.add_view(
             local_tracing_within_context, route_name="trace_context", renderer="json")
 
+        mock_filewatcher = mock.Mock(spec=FileWatcher)
+        mock_filewatcher.get_data.return_value = {
+            "secrets": {
+                "jwt/authentication/secret": {
+                    "type": "simple",
+                    "value": self.TOKEN_SECRET,
+                },
+            },
+            "vault": {
+                "token": "test",
+                "url": "http://vault.example.com:8200/",
+            }
+        }
+        secrets = store.SecretsStore("/secrets")
+        secrets._filewatcher = mock_filewatcher
```

Comment:
Rather than mocking out `secrets._filewatcher` it seems cleaner to me to mock out the SecretStore interface.  `AuthenticationContext` cares about the SecretStore but mocking out the FileWatcher makes these tests concerned about the inner workings of the SecretStore object rather than the interface it provides.